### PR TITLE
Script cleanups

### PR DESCRIPTION
**Pull request by pontushojer**
Cleaned up __main__.py, correctfastq.py and analyze.py scripts and made some style changes. Implemented dnaio for file reading and updated some logging and documentation. 

More extensive changes were made to analyze script where I changed the format of the output to be more intuitive and useful (Included header and barcodes). Also added ability to use the config files for naming the ABCs.

### CODE REVIEW

File: src/dbspro/cli/correctfastq.py [hunk 1]
```diff
@@ -2,64 +2,63 @@
...
+         dnaio.open(args.corr_fastq, mode="w", fileformat="fastq") as openout:
+        for read in tqdm(reader):
+
+            counter['tot_reads'] += 1
+            if read.sequence in err_corr:
+                read.sequence = err_corr[read.sequence]
+
+                # FIX: Possibly the sequence of the error corrected read is different from the original
+                if len(read.qualities) != len(read.sequence):
+                    read.qualities = "G"*len(read.sequence)
```
Comment (FrickTobias) [inline src/dbspro/cli/correctfastq.py]: Doesn't really matter. Could also change quality but we could also just write fasta as out.

File: src/dbspro/cli/analyze.py [hunk 2]
```diff
@@ -3,121 +3,143 @@
...
+    logger.info(f"Saving DBS information to RAM")
+
+    # Set names for ABCs. Creates dict with file names as keys and selected names as values.
+    default_tsv = os.path.join(os.path.dirname(__file__), "../../../construct-info/ABC-sequences.tsv")
+    if args.names == "Use file name":
+        abc_names = {file_name: file_name.split('/')[-1] for file_name in args.umi_abc}
+    elif args.names:
+        abc_names = get_names(args.names, args.umi_abc)
+    else:
+        abc_names = get_names(default_tsv, args.umi_abc)
```
[the remark this answers is not in the document]
Comment (pontushojer) [reply src/dbspro/cli/analyze.py]: Will make naming into new issue and do in separate PR.

File: src/dbspro/cli/analyze.py [hunk 3]
```diff
@@ -3,121 +3,143 @@
...
+                abc_counter_umi[abc].append(len(result_dict[bc][abc]))
+                # Add number of reads
+                abc_counter_read[abc].append(sum(result_dict[bc][abc].values()))
+
+        output_list.append(output_line)
+
+    # Create dataframe with barcode as index and columns with ABC data. Export to tsv.
+    df_out = pd.DataFrame(output_list, columns=["BC"] + sorted(abc_names.values())).set_index("BC", drop=True)
+    logging.info(f"Writing output file to: {args.output}")
+    df_out.to_csv(args.output, sep="\t")
```
[the remark this answers is not in the document]
Comment (pontushojer) [reply src/dbspro/cli/analyze.py]: I am not sure about the value of this, won't the new function simply do the same as this one line i have i.e.:
```python
df_out = pd.DataFrame(output_list, columns=["BC"] + sorted(abc_names.values())).set_index("BC", drop=True)
```

becomes

```python
df = build_df(output_list, columns=["BC"] + sorted(abc_names.values()), index="BC")


def build_df(some_list, columns=list()), index=None) :
   return pd.DataFrame(some_list, columns=columns).set_index(index, drop=True)
```

I think this function would only be more confusing. 

Or were you thinking something different?

File: src/dbspro/cli/correctfastq.py (hunk 1 above)
Comment (pontushojer) [reply src/dbspro/cli/correctfastq.py]: We talked about this and decided to leave it for now and make the file into a fasta instead in a later PR. I will make a new issue with this.

File: src/dbspro/cli/analyze.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (FrickTobias) [reply src/dbspro/cli/analyze.py]: I'm thinking rows 70 => 86 could be the function.

[the remark this answers is not in the document]
Comment (FrickTobias) [reply src/dbspro/cli/analyze.py]: Something like 
```
    # Output file writing and
    df_out = make_df_from_dict(dict=result_dict)
    
def make_df_from_dict(dict)
    """
    Builds pandas df from dictionary
    """

    output_list = list()
    for bc in iter(result_dict):
        output_line = {'BC': bc}

        for abc in abc_names.values():
            output_line[abc] = len(result_dict[bc][abc])

            # Add statistics if passing filter.
            if sum(result_dict[bc][abc].values()) >= args.filter:
                # Add number of UMI:s
                abc_counter_umi[abc].append(len(result_dict[bc][abc]))
                # Add number of reads
                abc_counter_read[abc].append(sum(result_dict[bc][abc].values()))

            output_list.append(output_line)

    df_out = build_df(output_list, columns=["BC"] + sorted(abc_names.values()), index="BC")

    return df_out

```

[the remark this answers is not in the document]
Comment (pontushojer) [reply src/dbspro/cli/analyze.py]: Yes fixed this in new commit. Had to also move the abc_counter into the function but its ok for now.

[the remark this answers is not in the document]
Comment (FrickTobias) [reply src/dbspro/cli/analyze.py]: Great, I'll go ahead and merge.

**Pull request merged**